There are some type errors:

```
[05:08:37 ](https://teamcity.a8c.com/buildConfiguration/calypso_calypso_WebApp_Run_All_Unit_Tests/13397803?buildTab=log&linesState=7898&logView=flowAware&focusLine=7898)
    client/components/stats-date-control/index.tsx(105,22): error TS7053: Element implicitly has an 'any' type because expression of type 'string' can't be used to index type '{ last_7_days: string; last_30_days: string; last_3_months: string; last_year: string; custom_date_range: string; apply_button: string; trigger_button: string; } | { last_7_days: string; ... 5 more ...; trigger_button: string; }'.

[05:08:37 ](https://teamcity.a8c.com/buildConfiguration/calypso_calypso_WebApp_Run_All_Unit_Tests/13397803?buildTab=log&linesState=7899&logView=flowAware&focusLine=7899)
      No index signature with a parameter of type 'string' was found on type '{ last_7_days: string; last_30_days: string; last_3_months: string; last_year: string; custom_date_range: string; apply_button: string; trigger_button: string; } | { last_7_days: string; ... 5 more ...; trigger_button: string; }'.

[05:08:37 ](https://teamcity.a8c.com/buildConfiguration/calypso_calypso_WebApp_Run_All_Unit_Tests/13397803?buildTab=log&linesState=7900&logView=flowAware&focusLine=7900)
    client/components/stats-date-control/index.tsx(114,22): error TS7053: Element implicitly has an 'any' type because expression of type 'string' can't be used to index type '{ last_7_days: string; last_30_days: string; last_3_months: string; last_year: string; custom_date_range: string; apply_button: string; trigger_button: string; } | { last_7_days: string; ... 5 more ...; trigger_button: string; }'.

[05:08:37 ](https://teamcity.a8c.com/buildConfiguration/calypso_calypso_WebApp_Run_All_Unit_Tests/13397803?buildTab=log&linesState=7901&logView=flowAware&focusLine=7901)
      No index signature with a parameter of type 'string' was found on type '{ last_7_days: string; last_30_days: string; last_3_months: string; last_year: string; custom_date_range: string; apply_button: string; trigger_button: string; } | { last_7_days: string; ... 5 more ...; trigger_button: string; }'.
```

--- a/client/components/stats-date-control/index.tsx
+++ b/client/components/stats-date-control/index.tsx
@@ -15,12 +15,28 @@ import './style.scss';
 const COMPONENT_CLASS_NAME = 'stats-date-control';
 const isCalendarEnabled = config.isEnabled( 'stats/date-picker-calendar' );
 
-// Hardcoding event names ensures consistency, searchability, and prevents errors per Tracks naming conventions.
-const eventNames = {
+// Define the event name keys for tracking events
+type EventNameKey =
+	| 'last_7_days'
+	| 'last_30_days'
+	| 'last_90_days'
+	| 'last_year'
+	| 'custom_date_range'
+	| 'apply_button'
+	| 'trigger_button';
+
+// Define the structure for tracking event names
+interface EventNames {
+	jetpack_odyssey: Record< EventNameKey, string >;
+	calypso: Record< EventNameKey, string >;
+}
+
+// Define the tracking event names object. Hardcoding event names ensures consistency, searchability, and prevents errors per Tracks naming conventions.
+const eventNames: EventNames = {
 	jetpack_odyssey: {
 		last_7_days: 'jetpack_odyssey_stats_date_picker_shortcut_last_7_days_click',
 		last_30_days: 'jetpack_odyssey_stats_date_picker_shortcut_last_30_days_click',
-		last_3_months: 'jetpack_odyssey_stats_date_picker_shortcut_last_3_months_click',
+		last_90_days: 'jetpack_odyssey_stats_date_picker_shortcut_last_90_days_click',
 		last_year: 'jetpack_odyssey_stats_date_picker_shortcut_last_year_click',
 		custom_date_range: 'jetpack_odyssey_stats_date_picker_shortcut_custom_date_range_click',
 		apply_button: 'jetpack_odyssey_stats_date_picker_apply_button_click',
@@ -29,7 +45,7 @@ const eventNames = {
 	calypso: {
 		last_7_days: 'calypso_stats_date_picker_shortcut_last_7_days_click',
 		last_30_days: 'calypso_stats_date_picker_shortcut_last_30_days_click',
-		last_3_months: 'calypso_stats_date_picker_shortcut_last_3_months_click',
+		last_90_days: 'calypso_stats_date_picker_shortcut_last_90_days_click',
 		last_year: 'calypso_stats_date_picker_shortcut_last_year_click',
 		custom_date_range: 'calypso_stats_date_picker_shortcut_custom_date_range_click',
 		apply_button: 'calypso_stats_date_picker_apply_button_click',
@@ -102,14 +118,14 @@ const StatsDateControl = ( {
 		const endDate = anchor.format( 'YYYY-MM-DD' );
 		const startDate = anchor.subtract( shortcut.range, 'days' ).format( 'YYYY-MM-DD' );
 
-		recordTracksEvent( eventNames[ event_from ][ shortcut.id ] );
+		recordTracksEvent( eventNames[ event_from ][ shortcut.id as EventNameKey ] );
 
 		// Update chart via routing.
 		setTimeout( () => page( generateNewLink( shortcut.period, startDate, endDate ) ), 250 );
 	};
 
 	// handler for shortcut clicks in new updated DateRange component
-	const onShortcutClickHandler = ( shortcutId: string ) => {
+	const onShortcutClickHandler = ( shortcutId: EventNameKey ) => {
 		const event_from = isOdysseyStats ? 'jetpack_odyssey' : 'calypso';
 		recordTracksEvent( eventNames[ event_from ][ shortcutId ] );
 	};

